src/pydvl/value/least_core/_common.py

```diff
@@ -1,108 +1,167 @@
+    :math:`A_{ub}` and :math:`A_{eq}` are matrices.
+
+    :param A_eq: The equality constraint matrix. Each row of ``A_eq`` specifies the
+        coefficients of a linear equality constraint on ``x``.
+    :param b_eq: The equality constraint vector. Each element of ``A_eq @ x`` must equal
+        the corresponding element of ``b_eq``.
+    :param A_lb: The inequality constraint matrix. Each row of ``A_lb`` specifies the
+        coefficients of a linear inequality constraint on ``x``.
+    :param b_lb: The inequality constraint vector. Each element represents a
+        lower bound on the corresponding value of ``A_lb @ x``.
+    :param options: Keyword arguments that will be used to select a solver
+        and to configure it. Refer to the following page for all possible options:
+        https://www.cvxpy.org/tutorial/advanced/index.html#setting-solver-options
     """
-    logger.debug(
-        f"Solving linear programming problem: {c=}, {A_eq=}, {b_eq=}, {A_ub=}, {b_ub=}"
-    )
-
-    result: scipy.optimize.OptimizeResult = scipy.optimize.linprog(
-        c,
-        A_ub=A_ub,
-        b_ub=b_ub,
-        A_eq=A_eq,
-        b_eq=b_eq,
-        bounds=bounds,
-        method="highs-ipm",
-        options=options,
-    )
+    logger.debug(f"Solving linear program : {A_eq=}, {b_eq=}, {A_lb=}, {b_lb=}")
+
+    x = cp.Variable(n_variables)
+    e = cp.Variable()
+    objective = cp.Minimize(e)
+    constraints = [
+        e >= 0,
+        A_eq @ x == b_eq,
+        A_lb @ x + e * np.ones(len(A_lb)) >= b_lb,
+    ]
+    problem = cp.Problem(objective, constraints)
+
+    solver = options.pop("solver", cp.ECOS)
+
+    try:
+        problem.solve(solver=solver, **options)
+    except cp.error.SolverError as err:
+        raise ValueError("Could not solve linear program") from err
+
+    if problem.status in cp.settings.SOLUTION_PRESENT:
+        logger.debug("Problem was solved")
+        if problem.status in [cp.settings.OPTIMAL_INACCURATE, cp.settings.USER_LIMIT]:
+            warnings.warn(
+                "Solver terminated early. Consider increasing the solver's "
+                "maximum number of iterations in options",
+                RuntimeWarning,
+            )
+        least_core_value = e.value.item()
+        # HACK: sometimes the returned least core value
+        # is negative but very close to 0
+        # to avoid any problems with the subsequent quadratic program
+        # we just set it to 0.0
+        if least_core_value < 0:
+            least_core_value = 0.0
+        return x.value, least_core_value
+
+    if problem.status in cp.settings.INF_OR_UNB:
+        warnings.warn(
+            "Could not find solution due to infeasibility or unboundedness of problem.",
+            RuntimeWarning,
+        )
+    return None, None
 
-    logger.debug(f"{result=}")
 
-    if result.success:
-        return np.asarray(result.x)
+def _solve_egalitarian_least_core_quadratic_program(
+    least_core_value: float,
```

Comment:
Value is already quite overloaded and the paper uses "subsidy", which makes sense
```suggestion
    least_core_subsidy: float,
```